src/fsspec_xrootd/xrootd.py

```diff
@@ -373,9 +373,9 @@ async def _rm_file(self, path: str, **kwargs: Any) -> None:
 
     async def _touch(self, path: str, truncate: bool = False, **kwargs: Any) -> None:
         if truncate or not await self._exists(path):
-            status, _ = await _async_wrap(self._myclient.truncate)(
-                path, size=0, timeout=self.timeout
-            )
+            f = client.File()
+            status, _ = f.open(path, OpenFlags.DELETE)
+            f.close()
```

Comment:
This is part of the `XRootDFileSystem` which is all asynchronous. So you'll need to await opening and then closing the file handle with ` await _async_wrap(f.open)(...)` and `await _async_wrap(f.close)(...)`

Reply:
changed